File: pubnub-gson/src/main/java/com/pubnub/api/endpoints/files/SendFile.java
package com.pubnub.api.endpoints.files;

import com.pubnub.api.endpoints.BuilderSteps;
import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.endpoints.files.requiredparambuilder.FilesBuilderSteps;
import com.pubnub.api.models.consumer.files.PNFileUploadResult;

public interface SendFile extends Endpoint<PNFileUploadResult> {

    SendFile message(Object message);

    SendFile meta(Object meta);

    SendFile ttl(Integer ttl);

    SendFile shouldStore(Boolean shouldStore);

    SendFile cipherKey(String cipherKey);

    interface Builder extends BuilderSteps.ChannelStep<FilesBuilderSteps.FileNameStep<FilesBuilderSteps.InputStreamStep<SendFile>>> {

    }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/kmp/endpoints/files/SendFile.kt
package com.pubnub.kmp.endpoints.files

import com.pubnub.api.models.consumer.files.PNFileUploadResult
import com.pubnub.kmp.PNFuture

/**
 * @see [PubNub.sendFile]
 */
interface SendFile : PNFuture<PNFileUploadResult>
